packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -157,6 +158,36 @@ typedef DelegatedTransitionBuilder =
       Widget? child,
     );
 
+mixin _RenderTransition<L extends Listenable> on RenderObject {
+  void _listener();
+
+  L get listenable => _listenable;
+  abstract L _listenable;
+  set listenable(L value) {
+    if (value == _listenable) {
+      return;
+    }
+    _listenable.removeListener(_listener);
+    _listenable = value;
+    if (attached) {
```

Comment:
Isn't there a bug then that if the listenable is changed while the RenderObject is unattached the `alignment` isn't updated properly and when the RO is attached it is rendered with a potentially outdated alignment value?